File: prepare_sanity_import.py
'''
import os
import json

def prepare_for_sanity():
    ndjson_path = 'import.ndjson'
    products_dir = 'import_products'

    with open(ndjson_path, 'w') as ndjson_file:
        for root, _, files in os.walk(products_dir):
            for file in files:
                if file == 'product.json':
                    file_path = os.path.join(root, file)
                    with open(file_path, 'r') as f:
                        try:
                            data = json.load(f)

                            # 1. Add _type
                            data['_type'] = 'product'

                            # 2. Use slug for _id
                            if 'slug' in data:
                                data['_id'] = data['slug']
                            else:
                                print(f"Skipping {file_path}, no slug found.")
                                continue
                            
                            # 3. Ensure price is a number
                            if 'price' in data and isinstance(data['price'], str):
                                try:
                                    data['price'] = int(data['price'])
                                except (ValueError, TypeError):
                                    print(f"Could not convert price to int for {file_path}. Removing price.")
                                    del data['price']
                            
                            # 4. Remove original image URLs, we will handle assets later
                            if 'images' in data:
                                del data['images']

                            ndjson_file.write(json.dumps(data) + '\n')

                        except json.JSONDecodeError:
                            print(f"Skipping {file_path} due to invalid JSON.")
                        except Exception as e:
                            print(f"An error occurred with {file_path}: {e}")
    print(f"Successfully created {ndjson_path}")

if __name__ == "__main__":
    prepare_for_sanity()
'''

File: prepare_sanity_import.js

const fs = require('fs');
const path = require('path');

const productsDir = 'import_products';
const ndjsonPath = 'import.ndjson';

// Ensure the output file is empty before we start
fs.writeFileSync(ndjsonPath, '');

function processDirectory(directory) {
    const items = fs.readdirSync(directory, { withFileTypes: true });

    for (const item of items) {
        const fullPath = path.join(directory, item.name);
        if (item.isDirectory()) {
            processDirectory(fullPath);
        } else if (item.name === 'product.json') {
            processProductFile(fullPath);
        }
    }
}

function processProductFile(filePath) {
    try {
        const fileContent = fs.readFileSync(filePath, 'utf-8');
        const data = JSON.parse(fileContent);

        // 1. Add _type
        data._type = 'product';

        // 2. Use slug for _id
        if (!data.slug) {
            console.warn(`Skipping ${filePath}, no slug found.`);
            return;
        }
        data._id = data.slug;

        // 3. Ensure price is a number
        if (data.price && typeof data.price === 'string') {
            const priceNum = parseInt(data.price, 10);
            if (!isNaN(priceNum)) {
                data.price = priceNum;
            } else {
                console.warn(`Could not convert price to int for ${filePath}. Removing price.`);
                delete data.price;
            }
        }

        // 4. Create brand reference
        if (data.brand) {
            data.brand = {
                _type: 'reference',
                _ref: `productBrand-${data.brand.toLowerCase().replace(/\s+/g, '-')}`
            };
        }

        // 5. Create category reference
        if (data.category) {
            data.category = {
                _type: 'reference',
                _ref: `productCategory-${data.category.toLowerCase().replace(/\s+/g, '-')}`
            };
        }

        // 6. Remove fields that are not needed or will be handled differently
        delete data.images;

        // Append the transformed object to the .ndjson file
        fs.appendFileSync(ndjsonPath, JSON.stringify(data) + '\n');

    } catch (error) {
        console.error(`An error occurred with ${filePath}:`, error);
    }
}

console.log('Starting preparation of Sanity import file...');
processDirectory(productsDir);
console.log(`Successfully created ${ndjsonPath}`);
